test/test_torch.py

```diff
@@ -12303,6 +12309,24 @@ def non_zero_rand(size, dtype, device):
                     non_zero_rand((2, 2), dtype=dtype, device=device),
                     non_zero_rand((2, 2), dtype=dtype, device=device))
 
+    # This function tests that a nan value is returned for input values not in domain
+    def test_acosh_domain(self, device):
+        # Domain of acosh is [1, inf)
+        sample = torch.tensor([-1.23, -0.06, 0.98], device=device)
```

Comment:
Please add -inf and +inf, to these values, too. Also you can extend this test with the @dtypes decorator to work on both torch.float32 and torch.complex64, right?

Reply:
Also 1, to be sure it's included.

Reply:
Thanks, @mruberry - adding `-inf` and `+inf` was a great idea as it led to this:

`std::acosh` returns `+inf` for `+inf` input. Verified this from https://en.cppreference.com/w/cpp/numeric/math/acosh (reliable) and also a test on my system.

This makes docs a little complicated, and I am open to any improvements in making the statements simple (if required).